### PR TITLE
[WIP][nanoGPT][DTensor] enable inplace dropout for nanoGPT benchmark

### PR DESCRIPTION
**Pull request by XilunWu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #104200

# WHY
non-inplace dropout seems not working with DTensor. need to further investigate.

### CODE REVIEW

File: torch/distributed/tensor/parallel/api.py [hunk 1]
```diff
@@ -94,6 +94,15 @@ def parallelize_module(  # type: ignore[return]
             return _parallelize_multihead_attn(module, device_mesh)
         elif _is_mlp_for_pairwise_parallel(module):
             return _parallelize_mlp(module, device_mesh, parallelize_plan)
+        elif isinstance(module, torch.nn.Dropout):
```
Comment (fduwjj) [inline torch/distributed/tensor/parallel/api.py:97]: I am wondering if directly inserting this logic in TP might be necessary or not.

My two cents are:
Let's make this optional for users or make this as an extension which user can pass in because what if today users all use pure DTensor ops or what if we want to have other special handling ops like this? (For example, softmax, layernorm, etc.)

**Review by fduwjj — COMMENTED** [review]
I am not against this change and benchmark work. But I feel we need to better think about how we merge this into our code base. Thanks!
